src/main/java/com/ozonehis/ozone_demo_data/service/DemoDataService.java

```diff
@@ -0,0 +1,155 @@
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class DemoDataService {
+
+    private static final String GENERATE_DEMO_DATA_ENDPOINT = "/ws/rest/v1/referencedemodata/generate";
+
+    private static final String KEYCLOAK_REALM = "ozone";
+
+    private static final int DEFAULT_DEMO_PATIENTS = 50;
+
+    private final SystemAvailabilityChecker systemAvailabilityChecker;
+
+    private final RestTemplate restTemplate;
+
+    private final OpenmrsConfig openmrsConfig;
+
+    private final KeycloakConfig keycloakConfig;
+
+    @Value("${openmrs.oauth.enabled:false}")
+    boolean oauthEnabled;
+
+    @Value("${number.of.demo.patients:" + DEFAULT_DEMO_PATIENTS + "}")
+    int numberOfDemoPatients;
+
+    public void triggerDemoData() {
+        try {
+            if (!systemAvailabilityChecker.waitForOpenMRSAvailability()) {
+                log.error("OpenMRS is not available. Aborting demo data generation.");
+                return;
+            }
+            triggerDemoDataGeneration();
+        } catch (Exception e) {
+            throw new DemoDataGenerationException("Failed to generate demo data", e);
+        }
+    }
+
+    private void triggerDemoDataGeneration() {
+        HttpHeaders headers = createAuthenticatedHeaders();
+        Map<String, Object> requestBody = createRequestBody();
+
+        HttpEntity<Map<String, Object>> request = new HttpEntity<>(requestBody, headers);
+        String generateDemoDataUrl = buildGenerateDemoDataUrl();
+
+        ResponseEntity<String> response = restTemplate.postForEntity(generateDemoDataUrl, request, String.class);
+
+        validateResponse(response);
+        log.info("Demo data generation completed successfully");
+    }
+
+    HttpHeaders createAuthenticatedHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        String authToken = oauthEnabled ? obtainOAuthToken() : obtainBasicAuthToken();
+
+        if (oauthEnabled) {
+            headers.setBearerAuth(authToken);
+        } else {
+            headers.setBasicAuth(authToken);
+        }
+
+        return headers;
+    }
+
+    private String obtainOAuthToken() {
+        log.info("OAuth2 authentication enabled. Obtaining OAuth token...");
+        try (Keycloak keycloak = createKeycloakClient()) {
+            return keycloak.tokenManager().getAccessToken().getToken();
+        } catch (Exception e) {
+            throw new AuthenticationException("Failed to obtain OAuth token", e);
+        }
+    }
+
+    private Keycloak createKeycloakClient() {
```

Comment:
What's the need of this?

```
@Bean
    public Keycloak keycloak() {
        return KeycloakBuilder.builder()
                .serverUrl(getServerUrl())
                .realm(getRealm())
                .grantType(OAuth2Constants.CLIENT_CREDENTIALS)
                .clientId(getClientId())
                .clientSecret(getClientSecret())
                .build();
    }
```

Reply:
Both does the same thing, need to remove one.